contracts/RiskManagerV1.sol

```diff
@@ -472,3 +661,249 @@ contract RiskManagerV1 is IRiskManager, Auctioneer, Ownable {
+/// @dev The client contract should take care of authorizations or governance
+///      delays according to their needs.
+library RiskManagerV1Rewards {
+    struct Storage {
+        // Fixed amount of COV tokens which should be given as reward for the
+        // notifier reporting about the start of deposit liquidation process.
+        uint256 liquidationNotifierRewardAmount;
+        uint256 newLiquidationNotifierRewardAmount;
+        uint256 liquidationNotifierRewardAmountChangeInitiated;
+        // Percentage of the total COV supply which should be given as reward
+        // for the notifier reporting about the start of deposit liquidation
+        // process.
+        uint256 liquidationNotifierRewardPercentage;
+        uint256 newLiquidationNotifierRewardPercentage;
+        uint256 liquidationNotifierRewardPercentageChangeInitiated;
+        // Fixed amount of COV tokens which should be given as reward for the
+        // notifier reporting about a deposit being liquidated outside of the
+        // coverage pool.
+        uint256 liquidatedNotifierRewardAmount;
+        uint256 newLiquidatedNotifierRewardAmount;
+        uint256 liquidatedNotifierRewardAmountChangeInitiated;
+        // Percentage of the total COV supply which should be given as reward
+        // for the notifier reporting about a deposit being liquidated outside
+        // of the coverage pool.
+        uint256 liquidatedNotifierRewardPercentage;
+        uint256 newLiquidatedNotifierRewardPercentage;
+        uint256 liquidatedNotifierRewardPercentageChangeInitiated;
+    }
+
+    event LiquidationNotifierRewardAmountUpdateStarted(
+        uint256 liquidationNotifierRewardAmount,
+        uint256 timestamp
+    );
+    event LiquidationNotifierRewardAmountUpdated(
+        uint256 liquidationNotifierRewardAmount
+    );
+
+    event LiquidationNotifierRewardPercentageUpdateStarted(
+        uint256 liquidationNotifierRewardPercentage,
+        uint256 timestamp
+    );
+    event LiquidationNotifierRewardPercentageUpdated(
+        uint256 liquidationNotifierRewardPercentage
+    );
+
+    event LiquidatedNotifierRewardAmountUpdateStarted(
+        uint256 liquidatedNotifierRewardAmount,
+        uint256 timestamp
+    );
+    event LiquidatedNotifierRewardAmountUpdated(
+        uint256 liquidatedNotifierRewardAmount
+    );
+
+    event LiquidatedNotifierRewardPercentageUpdateStarted(
+        uint256 liquidatedNotifierRewardPercentage,
+        uint256 timestamp
+    );
+    event LiquidatedNotifierRewardPercentageUpdated(
+        uint256 liquidatedNotifierRewardPercentage
+    );
+
+    /// @notice Begins the liquidation notifier reward amount update process.
+    /// @param _newLiquidationNotifierRewardAmount New liquidation notifier
+    ///        reward amount.
+    function beginLiquidationNotifierRewardAmountUpdate(
+        Storage storage self,
+        uint256 _newLiquidationNotifierRewardAmount
+    ) external {
```

Comment:
We can probably keep them all `internal`, the optimizer should copy the internal function code.

Reply:
But that will cause a growth of the `RiskManagerV1` bytecode size. Do we want that for sure?

Reply:
Done.